rllib/tests/test_env_with_subprocess.py

```diff
@@ -56,13 +56,11 @@ def leaked_processes():
             },
         },
     })
-    time.sleep(10.0)
     # Check whether processes are still running or Env has not cleaned up
     # the given tmp files.
-    leaked = leaked_processes()
-    assert not leaked, "LEAKED PROCESSES: {}".format(leaked)
-    assert not os.path.exists(tmp1), "atexit handler not called"
-    assert not os.path.exists(tmp2), "atexit handler not called"
-    assert not os.path.exists(tmp3), "close not called"
-    assert not os.path.exists(tmp4), "close not called"
+    wait_for_condition(lambda: not leaked_processes(), timeout=30)
```

Comment:
This is a max timeout, not a sleep, so it's fine. In general computers can sometimes take arbitrary long periods of time to respond under high load, so a max wait of 30s is pretty reasonable.

Reply:
But this means whenever we have test failures, the build takes 20 more seconds. This single test might have low impact, but if every test has 30 seconds timeout, for the same number of test failures it can have hours of differences. And that's why I changed the default timeout from 30 -> 10 before. I never really saw any issue that happened because of that change, so I prefer to just follow the default timeout unless there are special reasons (like this test won't pass if timeout is 10 seconds).

Reply:
Sang, 10 seconds is not enough time for any kind of process wait timeout and will itself cause flakiness. We should not be setting timeouts that low unless wrapped in a retry. In the common case, it will pass instantly add will not add any more time.